tests/lint/sconscript

```diff
@@ -0,0 +1,60 @@
+###
+# This file is a part of the NVDA project.
+# URL: http://www.nvaccess.org/
+# Copyright 2019 NV Access Limited.
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2.0, as published by
+# the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# This license can be found at:
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+###
+
+import os
+import sys
+
+import SCons
+
+Import("env")
+
+# Add system path to get access to git.
+externalEnv = Environment(
+	ENV = {'PATH' : os.environ['PATH']}
+)
+
+doLint = env.get("doLint")
+baseBranch = env.get("base")
+if doLint and not baseBranch:
+	errorMessage = (
+		"Lint can not complete without base branch. "
+		"Try: 'scons lint base=origin/master' "
+		"See also /tests/lint/readme.md"
+	)
+	raise SCons.SConf.SConfError(errorMessage)
+
+# Create a unified diff. Written to file for ease of manual inspection
+diffTarget = externalEnv.Command(

```

Comment:
Have you considered just importing the genDiff module and executing the diff creation from there? We use python calls python several times in our scons environment and I think it is ok, but it has its downsides.

Reply:
This needs to be run in the external environment, so that it has access to the path for git.